src/createElements.tsx

```diff
@@ -20,7 +20,7 @@ export default function createElements(
   routeMatches: Array<RouteMatch>,
   Components: React.ComponentType<any>[],
   matchData: any,
-): Array<ResolvedElement | undefined> {
+): Array<React.ReactNode | ResolvedElement | undefined> {
```

Comment:
I don't think this is the correct fix, This method _doesn't) actually return a ReactNode, it only turns route component elements. The problem is Redirect.render shouldn't return a ReactNode